lib/index.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+var EventEmitter = require('events').EventEmitter;
+var util = require('util');
+var ws = require('nodejs-websocket');
+
+function WSTransport () {
+	EventEmitter.call(this);
+	this._socket = null;
+}
+
+util.inherits(WSTransport, EventEmitter);
+
+WSTransport.register = function (transportProvider) {
+	transportProvider.registerTransport('wss', function () { return new WSTransport(); });
+};
+
+WSTransport.prototype.connect = function (address) {
+	this._socket = ws.connect(address.href, { extraHeaders: { 'Sec-Websocket-Protocol': 'AMQPWSB10' } });
```

Comment:
are these extra headers standard in some way? Might make sense to add a comment just in case. We can consider refactoring to allow e.g. specifying the headers in register and making sure they get passed to the constructor, but that seems overkill until we have another client.